sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -416,32 +416,31 @@ private[lf] object Speedy {
       * producing a text message.
       */
     private[speedy] override def handleException(excep: SValue.SAny): Control[Nothing] = {
-      @tailrec def unwind(): Option[KTryCatchHandler] =
+      @tailrec
+      def unwind(ptx: PartialTransaction): Option[KTryCatchHandler] =
         if (kontDepth() == 0) {
           None
         } else {
           popKont() match {
             case handler: KTryCatchHandler =>
```

Comment:
```suggestion
            case handler: KTryCatchHandler =>
            // The machine's ptx is updated even if the handler does not catch the exception.
            // This may cause the transaction trace to report the error from the handler's location.
            // This trade-off simplifies adding the trace to the exception directly.
```